### PR TITLE
Clean up client environment variables (and move one from client to server)

### PR DESCRIPTION
**Pull request by yxu-lanl**

### CODE REVIEW

File: installation/install.sh [hunk 1]
```diff
@@ -120,24 +120,35 @@ echo "web_server_domain=$web_server_domain" > $app_home/host.env
 echo "web_server_port=$web_server_port" >> $app_home/host.env
 
 #setup .env and server_pm2.json
+export LC_CTYPE=C
+oauth_secret=`cat /dev/urandom|tr -dc '[:alpha:]'|fold -w ${1:-20}|head -n 1`
+sendmail_key=`cat /dev/urandom|tr -dc '[:alpha:]'|fold -w ${1:-20}|head -n 1`
+jwt_key=`cat /dev/urandom|tr -dc '[:alpha:]'|fold -w ${1:-20}|head -n 1`
```
Comment (eecavanna) [inline installation/install.sh:126]: Will you add a comment saying what the command does?

Comment (eecavanna) [reply installation/install.sh:126]: I tried running the command on my Mac and it didn't run.

image

That is in zsh 5.9 running on a MacBook Pro M1 (MacOS Ventura 13.6).

Will you also document the system requirements for the command?

**Review by eecavanna — CHANGES_REQUESTED** [review]
Hi @yxu-lanl, thanks for eliminating the obsolete `REACT_APP_SOCIAL_LOGIN` environment variable and adding the `REACT_APP_SOCIAL_SECRET` one, as well as simplifying the placeholders for `JWT_KEY` and `SENDMAIL_KEY`.

I am confused about the command that, I assume, generates a random string. Will you add a comment that says what it does? Also, will you add a comment/documentation that says what the system requirements are for running those commands? They don't run on my Mac.

File: installation/install.sh (hunk 1 above)
Comment (yxu-lanl) [reply installation/install.sh:126]: 'export LC_CTYPE=C' is used to fix the 'tr: Illegal byte sequence' error on MacOS.

Comment (yxu-lanl) [reply installation/install.sh:126]: I tested the shell script on my MacOS and nmdc-edge Ubuntu.

Comment (eecavanna) [reply installation/install.sh:128]: Thanks!

**Review by eecavanna — APPROVED** [review]

**Comment by eecavanna** [comment]
Thanks for making these changes, @yxu-lanl.

**Comment by eecavanna** [comment]
Resolves https://github.com/microbiomedata/nmdc-edge/issues/35

**Pull request merged**